### PR TITLE
GSoC21: idea for Human Interface Guidelines Redesign

### LINKED ISSUE #112: Suggestion for GSoC 2021: Redesign Human Interface Guidelines and the Sugar UI
It would be nice to have colorful icons for sugar. This is one of the ideas which were not accepted last year into GSoC, for context: https://github.com/sugarlabs/GSoC/blob/master/Ideas-2020.md#colored-desktop-and-activity-icons

### PR DESCRIPTION
**Pull request by srevinsaju**
Closes #112 

This is a modified version of https://github.com/sugarlabs/GSoC/blob/master/Ideas-2020.md#colored-desktop-and-activity-icons, added in eaa46b15f2635bbbf07db89f0330d447686dffe1 by @walterbender.  Have made a few small improvements like considering the Sugar UI / UX along with icons and suggested a few optional tasks like finishing the left over design-based pull requests on `sugarlabs/sugar`

### CODE REVIEW

**Comment by quozl** [comment]
How many of the activities would be considered as completion?

**Comment by srevinsaju** [comment]
The Fructose set along with `sugar-artwork` would be a good start in my opinion. What do you think?

**Comment by quozl** [comment]
I agree, it would be a good start.  How to detect and handle activities that have not been converted?

**Comment by srevinsaju** [comment]
I would put forward a suggestion of adding 

```ini
IconColor = True
```
to `activity/activity.info` to those activities which have been converted. 

We could then add a commit to Sugar Shell to set the color of the icons which have `IconColor = False` to automatically use the colored icons instead of grey and white for the unused activities. If `IconColor = True` is set in the `activity.info` of the respective activity, then Sugar Shell should not attempt to give color to the activity icon, but instead use the default icon as defined in the activity.info..


In pseudo code
```python
if activity_info.is_color:
   # do not set color to the activity icon 
   # as the activity has been ported to multi colors
   # we should use the icon as is
   doNothingFunction()
else:
   # the activity did not set IconColor = True
   # this icon is an grey-white and follows the old Human Interface Guidelines
   # temporarily, we should set the icon to colorful based on the user selected color theme
   setActivityIconAsActive()
   


if activity_is_used:
   # add the badge for both types of icons
   addBadge()
```

Let me know if this explanation was okay. I am not sure if I could convey my idea properly :smile:

**Comment by quozl** [comment]
Or could we detect from the SVG or rendered pixmap that there are no colours other than black and white, and then assign two of the standard Sugar colours based on a hash of the bundle id?

**Comment by srevinsaju** [comment]
That would partially do it. I took a look at the Flappy Bird activity which was released to v4.activities.sugarlabs.org
image

In this case, the flappy icon is colorful. But yet, it needs a redesign.

**Comment by quozl** [comment]
https://github.com/sugarlabs/flappy/issues/14

File: Ideas-2021.md [hunk 1]
```diff
@@ -96,6 +97,79 @@ will help you to gain experience
...
+ - [ ] Add color functionality to the badges;
+ - [ ] Add badges in every instance where we currently use color: the
+       desktop, the journal, the neighborhood view and the activity
+       toolbar;
+ - [ ] Work with the design team to come up with new color icons for
+       all of the core Sugar toolbars, activities and activity toolbars
+       
+**Optional Tasks:**<br>
+ - [ ] Improve the Sugar Shell desktop UI, such as adding Animations 
+ - [ ] Port Sugar Toolkit to GTK4 (GTK4 has extensive support for animations)
```
Comment (JuiP) [inline Ideas-2021.md:155]: Can this optional task, be added as a separate project idea? I think it requires a greater time commitment and cannot be added as a task here.

**Review by JuiP — COMMENTED** [review]
Maybe you can consider adding [this](https://github.com/sugarlabs/GSoC/pull/114#issuecomment-774416841) to Project Task? It would help the student better understand the expected deliverables for the project.

File: Ideas-2021.md (hunk 1 above)
Comment (srevinsaju) [reply Ideas-2021.md:155]: Agreed. 

Btw, Did we receive the number of task allocated to Sugar Labs?

**Comment by quozl** [comment]
@srevinsaju asked
> Did we receive the number of tasks allocated to Sugar Labs?

Not yet.  See the timeline.  Approximate sequence is;

1. we make application to Google, (this is where we are up to),
2. Google select Sugar Labs,
3. Sugar Labs sets minimum and maximum slots,
4. Students make proposals,
5. Sugar Labs selects proposals,
6. Google agrees to specific number of slots,
7. Sugar Labs proceeds with specific proposals and students.

**Comment by srevinsaju** [comment]
@quozl said
> @srevinsaju asked
> 
> > Did we receive the number of tasks allocated to Sugar Labs?
> 
> Not yet. See the timeline. Approximate sequence is;
> 
> 1. we make application to Google, (this is where we are up to),
> 2. Google select Sugar Labs, (this is where our ideas may be reviewed by Google),
> 3. Sugar Labs sets minimum and maximum slots,
> 4. Students make proposals, (this is where our ideas may be reviewed by students),
> 5. Sugar Labs selects proposals,
> 6. Google agrees to specific number of slots, (this is where our ideas may be reviewed by Google again),
> 7. Sugar Labs proceeds with specific proposals and students.  (this is where our ideas not only have to be doable, but we have to be ready to get started on them with the student).

Thanks for the reply. 


@JuiP said
> Maybe you can consider adding [this](https://github.com/sugarlabs/GSoC/pull/114#issuecomment-774416841) to Project Task? It would help the student better understand the expected deliverables for the project.

I would not add that yet. I will add that after #118 has been decided. I am interested to see the student who is willing to do the port to GTK4 to implement the badges for the activities, as the port and the badges feature would otherwise conflict. If #118 is approved, I would prefer to remove the badge feature from #114 and increase the number of activities from Fructose set to something more.

**Comment by quozl** [comment]
Thanks for progressing this.  

Deadline is Friday, Feb 19th at 1900 UTC.

File: Ideas-2021.md (hunk 1 above)
Comment (srevinsaju) [reply Ideas-2021.md:159]: Done as of https://github.com/sugarlabs/GSoC/pull/114/commits/309366c3a69127889c93be3d12f1058506af673d

**Comment by srevinsaju** [comment]
> Thanks for progressing this.
> 
> Deadline is Friday, Feb 19th at 1900 UTC.

Thanks, updated the PR. Kindly review

**Comment by quozl** [comment]
Reviewed.  I think it would be helpful for a student to know that the reason for monochrome icons and high contrasting colour choices was to support reflective displays that cannot show colour.  In turn that was because backlights could not be made bright enough for use in high ambient light, without risking large amounts of incidental heat.  Since then backlights are more efficient (less heat) and much brighter (in nits).  Now that the reflective display technology has all but been abandoned, the requirement is removed, and so Sugar can go for a UI design that is more colourful.

File: Ideas-2021.md (hunk 1 above)
Comment (Deepak-Chaurasiya) [reply Ideas-2021.md:159]: Sir can i start working on the project

Comment (quozl) [reply Ideas-2021.md:159]: We are a permissionless community; you don't need to ask for permission, you can start whatever you like.  As far as Sugar Labs is concerned, if you finish this before we start GSoC, we get what we want, and we won't need to run the project.  If you get a good start on it, and then make a proposal as a GSoC student, then that will demonstrate your capacity, and we will take it into account.  If there are others who also build on your work, they may achieve still greater things.  If you aren't even hoping to be part of GSoC, then that's fine too.  Sugar Labs gets what we need.  :grin:

**Comment by chimosky** [comment]
@srevinsaju any other thing you want to add to this besides what @quozl mentioned above?

**Comment by srevinsaju** [comment]
@chimosky  No, nothing else.

**Comment by srevinsaju** [comment]
@quozl @chimosky @JuiP added all the required changes. Kindly review .

**Comment by quozl** [comment]
I approve.  @JuiP as org admin, please review.

**Comment by JuiP** [comment]
reviewed; looks good to go.
@srevinsaju could you please rebase?

**Comment by srevinsaju** [comment]
Rebased.

**Comment by JuiP** [comment]
Thanks! Merged :smile:

**Pull request merged**